lib/HLSL/DxilContainerReflection.cpp

```diff
@@ -1813,14 +1861,14 @@ HRESULT DxilShaderReflection::GetDesc(D3D12_SHADER_DESC *pDesc) {
   // Unset:  UINT                    ArrayInstructionCount;       // Number of array instructions used
   // Unset:  UINT                    CutInstructionCount;         // Number of cut instructions used
   // Unset:  UINT                    EmitInstructionCount;        // Number of emit instructions used
-  // Unset:  D3D_PRIMITIVE_TOPOLOGY  GSOutputTopology;            // Geometry shader output topology
-  // Unset:  UINT                    GSMaxOutputVertexCount;      // Geometry shader maximum output vertex count
-  // Unset:  D3D_PRIMITIVE           InputPrimitive;              // GS/HS input primitive
-  // Unset:  UINT                    cGSInstanceCount;            // Number of Geometry shader instances
-  // Unset:  UINT                    cControlPoints;              // Number of control points in the HS->DS stage
-  // Unset:  D3D_TESSELLATOR_OUTPUT_PRIMITIVE HSOutputPrimitive;  // Primitive output by the tessellator
-  // Unset:  D3D_TESSELLATOR_PARTITIONING HSPartitioning;         // Partitioning mode of the tessellator
-  // Unset:  D3D_TESSELLATOR_DOMAIN  TessellatorDomain;           // Domain of the tessellator (quad, tri, isoline)
+  pDesc->GSOutputTopology = ToD3DEnum(M.GetStreamPrimitiveTopology());
+  pDesc->GSMaxOutputVertexCount = M.GetMaxVertexCount();
+  pDesc->InputPrimitive = (D3D_PRIMITIVE)M.GetInputPrimitive();
```

Comment:
Question for @tex3d : Some code higher up assumes that these enums can be casted, but they are not documented as matching the corresponding D3D enums so I preferred using switch statements. Should I change this one to another `ToD3DEnum` call, or change all of them to straight casts?

```
D3D_PRIMITIVE DxilShaderReflection::GetGSInputPrimitive() {
  if (!m_pDxilModule->GetShaderModel()->IsGS())
    return D3D_PRIMITIVE::D3D10_PRIMITIVE_UNDEFINED;
  return (D3D_PRIMITIVE)m_pDxilModule->GetInputPrimitive();
} 
```

Reply:
A comment should be added to these enums in DxilConstants.h to make it clear that they must match the D3D enums.  They are explicitly set to the same values so they can be cast.  This should allow you to remove translation functions above.

Here are some applicable enums that do not currently have the comment:
DXIL::ShaderKind == D3D11_SHADER_VERSION_TYPE
DXIL::InterpolationMode == D3D_INTERPOLATION_MODE
DXIL::InputPrimitive == D3D_PRIMITIVE
DXIL::PrimitiveTopology == D3D_PRIMITIVE_TOPOLOGY
DXIL::TessellatorDomain == D3D_TESSELLATOR_DOMAIN
DXIL::TessellatorPartitioning == D3D_TESSELLATOR_PARTITIONING
DXIL::TessellatorOutputPrimitive == D3D_TESSELLATOR_OUTPUT_PRIMITIVE

Note: some enums may be lacking values on one side or the other.  This is OK, as some values are used on one side but not the other.  They must agree on the values they both define.

However, for InputPrimitive, I believe you must translate for hull shader case `if (...->IsHS())` using:
`pDesc->InputPrimitive = (D3D_PRIMITIVE)(D3D_PRIMITIVE_1_CONTROL_POINT_PATCH + m_GetInputControlPointCount() - 1);`